index.js

```diff
@@ -115,7 +120,8 @@ class Transport { //jshint ignore:line
         if (opts.text.length > 4000) {
             opts.text = opts.text.substr(0, 4000);
         }
-        req.write(JSON.stringify(opts));
+        // override aggregation_key with version passed as param (if set)
+        req.write(JSON.stringify(Object.assign({ aggregation_key: aggregation_key }, opts)));
```

Comment:
why not set it similarly to how the rest of the options are being set, to maintain consistency with the rest of the code:

```
if (aggregation_key) {
  opts.aggregation_key = aggregation_key
}
```

Reply:
`opts` is a reference to the Transport Options. Doing it this way makes a copy with the aggregation key set to the passed in parameter, without updating the value set on the Transport Options (if any).